lib/jazz_fingers/prompt.rb

```diff
@@ -28,48 +41,81 @@ def bold_text(text)
       "\001\e[1m\002#{text}\001\e[0m\002"
     end
 
-    def separator
+    def main_separator
       red_text(@separator)
     end
 
-    def name
-      blue_text(@application_name)
+    def wait_separator
+      "*"
+    end
+
+    # Return the current Pry context
+    #
+    # When the Pry context is `"main"` or `"nil"`, use the application name from
+    # the JazzFingers config. Examples: "(my_rails_app_name)", "(jazz_fingers)".
+    #
+    # When in the context of an object instance, use the abbreviated object
+    # path. Example: "(#<Pry::Prompt>)", "(#<...::ModuleName::ClassName>)"
+    #
+    # Fall back to the raw context provided by Pry.view_clip.
+    # Example: "(Pry::Prompt)"
+    def context(module_name = "main")
+      name =
+        case module_name
+        when "main", "nil"
+          @application_name
+        when OBJECT_INSTANCE
+          abbreviated_context(module_name)
+        else
+          module_name
+        end
+
+      blue_text("(#{name})")
     end
 
     def line_number(pry)
       if pry.respond_to? :input_ring
-        "[#{bold_text(pry.input_ring.size)}]"
+        bold_text(pry.input_ring.size)
       else
-        "[#{bold_text(pry.input_array.size)}]"
+        bold_text(pry.input_array.size)
       end
     end
 
-    def text(object, level)
-      level = 0 if level < 0
-      text = Pry.view_clip(object)
+    # Abbreviate the object path for object instances to a maximum length of
+    # `max_length` so the prompt doesn't overflow.
+    #
+    # Examples:
+    #   In:  #<Class1::Class2::Class3::Class4::Class5>
+    #   Out: #<...Class2::Class3::Class4::Class5>
+    #
+    #   In:  #<Class1::Class2>
+    #   Out: #<Class1::Class2>
+    #
+    #   In:  #<NoPathJustASingleLongClassName>
+    #   Out: #<NoPathJustASingleLongClassName>
+    def abbreviated_context(object_label, max_length: 20)
+      object_path = object_label[OBJECT_INSTANCE, 1]
+      object_path_components = object_path.split("::")
+      return object_label if object_path_components.length == 1
 
-      if text == 'main'
-        ''
-      else
-        "(#{'../' * level}#{text})"
-      end
-    end
+      object_path_length = object_path_components.map(&:length).sum
+      return object_label if object_path_length < max_length
```

Comment:
If we follow my last comment i believe this should be the first guard clause on this method.